### ISSUE TITLE
QuestieFramePool.lua line 557: attempt to index field 'data' (a nil value)

### ISSUE THREAD

**Issue by drejjmit** [AeroScripts/QuestieDev#1584]
## Bug description
When questing in Duskwood:

> Date: 2019-11-23 12:59:20
> ID: 1
> Error occured in: Global
> Count: 1
> Message: ...ddOns\Questie\Modules\FramePool\QuestieFramePool.lua line 557:
>    attempt to index field 'data' (a nil value)
> Debug:
>    ...ddOns\Questie\Modules\FramePool\QuestieFramePool.lua:557:
>       ...ddOns\Questie\Modules\FramePool\QuestieFramePool.lua:527
> Locals:
> self = QuestieFrame245 {
>  0 = <userdata>
>  glow = QuestieFrame245Glow {
>  }
>  BaseOnUpdate = <function> defined @Interface\AddOns\Questie\Modules\FramePool\QuestieFrame.lua:188
>  FakeUnhide = <function> defined @Interface\AddOns\Questie\Modules\FramePool\QuestieFrame.lua:350
>  FakeHide = <function> defined @Interface\AddOns\Questie\Modules\FramePool\QuestieFrame.lua:334
>  texture = <unnamed> {
>  }
>  BaseOnShow = <function> defined @Interface\AddOns\Questie\Modules\FramePool\QuestieFrame.lua:194
>  glowLogicTimer = <table> {
>  }
>  BaseOnHide = <function> defined @Interface\AddOns\Questie\Modules\FramePool\QuestieFrame.lua:218
>  FadeOut = <function> defined @Interface\AddOns\Questie\Modules\FramePool\QuestieFrame.lua:306
>  UpdateTexture = <function> defined @Interface\AddOns\Questie\Modules\FramePool\QuestieFrame.lua:222
>  glowTexture = <unnamed> {
>  }
>  Unload = <function> defined @Interface\AddOns\Questie\Modules\FramePool\QuestieFrame.lua:257
>  GlowUpdate = <function> defined @Interface\AddOns\Questie\Modules\FramePool\QuestieFrame.lua:170
>  FadeIn = <function> defined @Interface\AddOns\Questie\Modules\FramePool\QuestieFrame.lua:320
>  frameId = 245
> }
> r = 0.999998
> g = 0.999998
> b = 0.999998
> a = 0.999998
> Tooltip = GameTooltip {
>  0 = <userdata>
>  SetQuestLogRewardSpell = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  SetCurrencyByID = <function> defined @Interface\AddOns\BankItems\BankItems.lua:5904
>  SetCurrencyToken = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  SetTradeSkillItem = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  hasMoney = 1
>  SetMerchantItem = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  manualOrder = <table> {
>  }
>  questOrder = <table> {
>  }
>  SetCraftItem = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  SetQuestRewardSpell = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  canAddRestStateLine = 1
>  comparing = false
>  TopOverlay = <unnamed> {
>  }
>  SetPetAction = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  SetHyperlink = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  SetShapeshift = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  SetLootItem = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  SetQuestItem = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  _owner = QuestieFrame245 {
>  }
>  SetTrainerService = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  numMoneyFrames = 1
>  SetQuestCurrency = <function> defined @Interface\AddOns\BankItems\BankItems.lua:5962
>  Show = <function> defined =[C]:-1
>  SetInboxItem = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  SetBuybackItem = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  shoppingTooltips = <table> {
>  }
>  npcOrder = <table> {
>  }
>  SetCurrencyTokenByID = <function> defined @Interface\AddOns\BankItems\BankItems.lua:5924
>  SetQuestLogItem = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  SetBackpackToken = <function> defined @Interface\AddOns\BankItems\BankItems.lua:5939
>  SetTradeTargetItem = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  SetTalent = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  SetTradePlayerItem = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  SetGuildBankItem = <function> defined =[C]:-1
>  SetSendMailItem = <function> defined @Interface\AddOns\Auc-Advanced\Libs\LibExtraTip\LibExtraTip.lua:418
>  SetLFGDungeonReward = <function> defined @Interface\AddOns\BankItems\BankItems.lua:6002
>  updateTooltip = 0.196000
>  SetInventoryItem = <function> defined @Interface\

```[13:07:47] QuestieFrame245={
[13:07:47]   [0]=<userdata QuestieFrame245[0]>,
[13:07:47]   glow={
[13:07:47]     [0]=<userdata QuestieFrame245Glow[0]>
[13:07:47]   },
[13:07:48]   BaseOnUpdate=<function>,
[13:07:48]   FakeUnhide=<function>,
[13:07:48]   FakeHide=<function>,
[13:07:48]   texture={
[13:07:48]     SetVertexColor=<function>,
[13:07:48]     a=1,
[13:07:48]     r=1,
[13:07:48]     b=1,
[13:07:48]     OLDSetVertexColor=<function>,
[13:07:48]     g=1,
[13:07:48]     [0]=<userdata>
[13:07:48]   },
[13:07:48]   BaseOnShow=<function>,
[13:07:48]   glowLogicTimer={
[13:07:48]     _callback=<function>,
[13:07:48]     _cancelled=true
[13:07:48]   },
[13:07:48]   BaseOnHide=<function>,
[13:07:48]   FadeOut=<function>,
[13:07:48]   UpdateTexture=<function>,
[13:07:48]   glowTexture={
[13:07:48]     SetVertexColor=<function>,
[13:07:48]     a=0.99999779462814,
[13:07:48]     r=0.59114810632649,
[13:07:48]     b=0.88316446424757,
[13:07:48]     OLDSetVertexColor=<function value>,
[13:07:48]     g=0.85618610187078,
[13:07:48]     [0]=<userdata>
[13:07:48]   },
[13:07:48]   Unload=<function>,
[13:07:48]   GlowUpdate=<function>,
[13:07:48]   FadeIn=<function>,
[13:07:48]   frameId=245
[13:07:48] }
```

## Screenshots
![image](https://user-images.githubusercontent.com/15526476/69478323-67531900-0df1-11ea-994c-20fa0b63117a.png)

Works after reload. Difference:
![image](https://user-images.githubusercontent.com/15526476/69478542-eb0e0500-0df3-11ea-9fb1-3704e6830512.png)

## Questie version
v5.2.3

**Comment by BreakBB** [comment]
This is related to nodes not getting removed correctly. There is also a commend for this:

```lua
-- FIXME: `data` can be nil here which leads to an error, will have to debug:
    -- https://discordapp.com/channels/263036731165638656/263040777658171392/627808795715960842
    -- happens when a note doesn't get removed after a quest has been finished, see #1170
    -- TODO: change how the logic works, so this [ObjectiveIndex?] can be nil
    -- it is nil on some notes like starters/finishers, because its for objectives. However, it needs to be an integer here for duplicate checks
```

Will be irrelevant once @Logonz finished his rewriting of the map logic.

**Comment by BreakBB** [comment]
Fixed in v5.7.2
